test/dist_autograd_test.py

```diff
@@ -4,22 +4,46 @@
 import unittest
 
 import torch
+import torch.distributed as dist
 import torch.distributed.autograd as dist_autograd
 import torch.distributed.rpc as rpc
 from dist_utils import INIT_METHOD_TEMPLATE, dist_init
 
 import threading
 
-prev_rank_rpc_done = False
 prev_rank_context_id = 0
 
+known_context_ids = []
+
+# we don't need a lock here since the GIL is held while executing remote
+# python UDFs, so access to known_context_ids is serialized across several workers.
+def store_context_id(context_id):
+    global known_context_ids
+    known_context_ids.append(context_id)
 
 def _set_rpc_done(context_id):
-    global prev_rank_rpc_done
     global prev_rank_context_id
-    prev_rank_rpc_done = True
     prev_rank_context_id = context_id
 
+# after dist autograd context is cleaned up, it should be cleaned up on other
+# nodes. This helper allows timeout_seconds for those RPCs to be completed, and
+# ensures that all the contexts have been cleaned up in that timeframe.any
+def _all_contexts_cleaned_up(num_contexts, timeout_seconds=10):
```

Comment:
@pritamdamania87 I implemented your suggestion and removed the use of `rpc.sync_rpc()` - is this 10s timeout ok or will we need a longer one for TSAN/ASAN tests?

Reply:
10s should be sufficient.